src-plugins/vtkDataMesh/readers/vtkDataMeshReader.cpp

```diff
@@ -86,140 +85,110 @@ bool vtkDataMeshReader::read(const QString& path) {
-            parseHeaderVtk(header, medData);
-        }
-        else if (extension == "vtp") // VTP files
+        try
         {
             dataSet->Read(path.toLocal8Bit().constData());
             medData->setData(dataSet);
 
-            // Extract field data from the xml file
-            vtkSmartPointer<vtkXMLPolyDataReader> reader = vtkSmartPointer<vtkXMLPolyDataReader>::New();
-            reader->SetFileName(path.toLocal8Bit().constData());
-            reader->Update();
-
-            if (reader->GetOutput()->GetFieldData())
+            if (!extractMetaData(path, dataSet))
             {
-                parseHeaderVtp(reader->GetOutput()->GetFieldData(), medData);
+                qDebug() << metaObject()->className() << ": no metadata found in " << path;
             }
+
+            setProgress(100);
+            return true;
         }
-        else if ((extension == "mesh") || (extension == "obj")) // MESH or OBJ files
+        catch (vtkErrorCode::ErrorIds error)
         {
-            try
-            {
-                dataSet->Read(path.toLocal8Bit().constData());
-            } catch (...)
-            {
-                qDebug() << "vtkDataMeshReader::read -> loading the vtkDataMesh failed";
-                return false;
-            }
+            qDebug() << metaObject()->className() << ": " << vtkErrorCode::GetStringFromErrorCode(error);
+            return false;
         }
-        else
+        catch (...)
         {
+            qDebug() << metaObject()->className() << ": error reading " << path;;
             return false;
         }
     }
-
-    setProgress(100);
-    return true;
+    else
+    {
+        return false;
+    }
 }
 
-QStringList vtkDataMeshReader::metaDataKeysToCopy()
+bool vtkDataMeshReader::extractMetaData(QString path, vtkMetaDataSet* dataSet)
 {
-    QStringList keys;
-
-    keys << medMetaDataKeys::PatientID.key()
-         << medMetaDataKeys::PatientName.key()
-         << medMetaDataKeys::Age.key()
-         << medMetaDataKeys::BirthDate.key()
-         << medMetaDataKeys::Gender.key()
-         << medMetaDataKeys::Description.key()
-         << medMetaDataKeys::StudyID.key()
-         << medMetaDataKeys::StudyDicomID.key()
-         << medMetaDataKeys::StudyDescription.key()
-         << medMetaDataKeys::Institution.key()
-         << medMetaDataKeys::Referee.key()
-         << medMetaDataKeys::StudyDate.key()
-         << medMetaDataKeys::StudyTime.key()
-         << medMetaDataKeys::Performer.key()
-         << medMetaDataKeys::Report.key()
-         << medMetaDataKeys::Protocol.key()
-         << medMetaDataKeys::Origin.key()
-         << medMetaDataKeys::AcquisitionDate.key()
-         << medMetaDataKeys::AcquisitionTime.key()
-         << medMetaDataKeys::Modality.key()
-         << medMetaDataKeys::Orientation.key();
-
-    return keys;
+    if (extractMetaDataFromFieldData(dataSet) || extractMetaDataFromHeader(path, dataSet) || extractCartoMetaData(dataSet))
```

Comment:
The problem has nothing to do with splitting up the ```if``` statement because it does the exact same thing. You simply got a different result because you changed the order of the calls. It won't work for all cases though because of the way the functions were written (```extractMetaDataFromHeader``` is meant to be called before the CARTO version).
This was working last time, so I might have messed up while pushing the code. I'll check on it.